packages/studio-base/src/providers/LayoutManagerProvider.tsx

```diff
@@ -33,6 +39,31 @@ export default function LayoutManagerProvider({ children }: React.PropsWithChild
     layoutManager.setOnline(online);
   }, [layoutManager, online]);
 
+  useEffect(() => {
+    if (loaders.length === 0) {
+      return;
+    }
+    const loadAndSaveLayouts = async () => {
+      try {
+        const currentLayouts = await layoutManager.getLayouts();
+        loaders.forEach(async (loader) => {
+          for (const layout of await loader.fetchLayouts()) {
```

Comment:
Maybe using map instead of two for loops can make it more readble and easier to mantain. We can also add one more step of error handling.
This is an example (needs to be tested):
`const loaderPromises = loaders.map(async (loader) => {
          try {
            const fetchedLayouts = await loader.fetchLayouts();
            return fetchedLayouts.filter(layout => !currentLayoutFroms.has(layout.from));
          } catch (loaderError) {
            log.error(`Failed to fetch layouts from loader: ${loaderError}`);
            return [];
          }
        });`

Reply:
It was rewritten with your suggestions. Indeed using Promise.allSettled give more space to catch individual errors and does not break all the code if the first failed, for instance. I think it increases a lot the processing (doing and redoing a lot of maps and filters), but this functionality will run only one time and it is very hard to think that the user could have millions of layouts, so I guess readability and error handling is really more important here. Good catch!